### DISCUSSION TITLE
How to create/add a new index with rangeKey?

### DISCUSSION THREAD

**Discussion by p32929** [dynamoose/dynamoose#1471] [Q&A] [answered]
Hi, thanks for creating and maintaining dynamoose. My question is:

Lets say, we have a model like this:
```
export const SomethingSchema = new Schema(
    {
        id: {
            type: String,
            hashKey: true,
        },
        text: {
            type: String,
        },
    },
    {
        timestamps: true,
    },
);
```
& Now, we want to make/change it like this:

```
export const SomethingSchema = new Schema(
    {
        id: {
            type: String,
            hashKey: true,
        },
        text: {
            type: String,
        },
        byCreatedAt: {
            type: String,
            default: "byCreatedAt",
            index: {
                global: true,
                rangeKey: 'createdAt',
                name: 'createdAtIndex',
                project: true,
                throughput: 5,
            }
        },
    },
    {
        timestamps: true,
    },
);
```
We basically added a new attribute and an index.
Now, if we add some data to the table and try to query it like this:
```
somethingModel.query("byCreatedAt").eq("byCreatedAt").exec()
```
we get errors like:
```
ValidationException: The table does not have the specified index: createdAtIndex
```

How to handle this error?
Thanks

**Comment by andrewda** [comment, marked as answer]
That `ValidationExcepton` is coming directly from the AWS SDK. Does the `createdAtIndex` exist on your table? If not, you can [create it manually on the AWS Console](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/getting-started-step-6.html), or pass `update: true` (or `update: ['indexes']`) in the [Table config](https://dynamoosejs.com/guide/Table#new-dynamoosetablename-models-options) to tell Dynamoose to create it automatically. Note that for bigger tables, it may take a while for the index to fully backfill and be ready for use.

**Reply by andrewda** [reply]
All good! Indexes won't be created automatically by DynamoDB, so you'll need to use one of the methods I mentioned above. I recommend reading the [DynamoDB docs on indexes](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/SecondaryIndexes.html) to become familiar with how they work, since they can be a little confusing and differ slightly from most SQL databases.

**Reply by p32929** [reply]
Thanks a lot. The issue is resolved.

**Reply by p32929** [reply]
BTW, when I try to create multiple global index, like this code:
```
export const EntitySchema = new Schema(
  {
    id: {
      type: String,
      hashKey: true,
    },
    byCreatedAt: {
      type: String,
      default: "byCreatedAt",
      forceDefault: true,
      index: {
        global: true,
        rangeKey: 'createdAt',
        name: 'createdAtIndex',
        project: true,
      }
    },
    byName: {
      type: String,
      default: "byName",
      forceDefault: true,
      index: {
        global: true,
        rangeKey: 'name',
        name: 'byNameIndex',
        project: true,
      }
    },
    byEmail: {
      type: String,
      default: "byEmail",
      forceDefault: true,
      index: {
        global: true,
        rangeKey: 'email',
        name: 'byEmailIndex',
        project: true,
      }
    },
    byPhone: {
      type: String,
      default: "byPhone",
      forceDefault: true,
      index: {
        global: true,
        rangeKey: 'phone',
        name: 'byPhoneIndex',
        project: true,
      }
    },
    name: {
      type: String,
    },
    phone: {
      type: String,
    },
    email: {
      type: String,
    },
  },
  {
    timestamps: true,
  },
);

```

It only creates one index and doesn't create the other indexes. Even though, I have set `update` to `true`. Any way to create multiple indexes in one go?
Thanks
